x/ecocredit/base/keeper/features/msg_update_project_enrollment.feature

```diff
@@ -0,0 +1,95 @@
+        | Bob unspecified to accepted       | UNSPECIFIED       | abc          | Bob    | ACCEPTED   | foo123       | unauthorized | UNSPECIFIED       | abc               |
+        | I01 changes requested to accepted | CHANGES_REQUESTED | bar456       | I01    | ACCEPTED   | foo123       |              | ACCEPTED          | foo123            |
+        | Bob changes requested to accepted | CHANGES_REQUESTED | bar456       | Bob    | ACCEPTED   | foo123       | unauthorized | CHANGES_REQUESTED | bar456            |
+        | I01 update accepted metadata      | ACCEPTED          | foo123       | I01    | ACCEPTED   | bar357       |              | ACCEPTED          | bar357            |
+        | Bob updated accepted metadata     | ACCEPTED          | foo123       | Bob    | ACCEPTED   | bar357       | unauthorized | ACCEPTED          | foo123            |
+
+  Rule: valid state transitions performed by issuers which remove enrollment entries are:
+  UNSPECIFIED -> REJECTED
+  CHANGES_REQUESTED -> REJECTED
+  ACCEPTED -> TERMINATED
+    Scenario Outline: <scenario>
+      Given enrollment for "P001" to "C01" is "<cur_status>" with metadata "<cur_metadata>"
+      When "<issuer>" updates enrollment for "P001" to "C01" with status "<new_status>" and metadata "<new_metadata>"
+      Then expect error contains "<err>"
+      And expect enrollment exists for "P001" to "C01" to be "<exists>"
+      And if no error expect EventUpdateProjectEnrollment with properties
+      """
+        {
+            "issuer": "<issuer>", "project_id": "P001", "class_id": "C01",
+            "old_status": "PROJECT_ENROLLMENT_STATUS_<cur_status>",
+            "new_status": "PROJECT_ENROLLMENT_STATUS_<new_status>",
+            "new_enrollment_metadata": "<new_metadata>"
+        }
+      """
+      Examples:
+        | scenario                            | cur_status        | cur_metadata | issuer | new_status | new_metadata | err          | exists |
+        | IO1 unspecified to rejected         | UNSPECIFIED       | abc          | I01    | REJECTED   | baz789       |              | false  |
+        | Bob unspecified to rejected         | UNSPECIFIED       | abc          | Bob    | REJECTED   | baz789       | unauthorized | true   |
+        | I01 changes requested to rejected   | CHANGES_REQUESTED | bar456       | I01    | REJECTED   | baz789       |              | false  |
+        | Bob changes requested to rejected   | CHANGES_REQUESTED | bar456       | Bob    | REJECTED   | baz789       | unauthorized | true   |
+        | I01 accepted to rejected            | ACCEPTED          | foo123       | I01    | REJECTED   | baz789       | invalid      | true   |
+        | Bob accepted to rejected            | ACCEPTED          | foo123       | Bob    | REJECTED   | baz789       | unauthorized | true   |
```

Comment:
Related to my other comment, I think this is the only test for an invalid state transition from enrollment status `ACCEPTED`.

Reply:
Added more tests. Let me know if that looks good enough

Reply:
Yep, looks good!